### ISSUE TITLE
"space-in-brackets" inconsistency?

### ISSUE THREAD

**Issue by bytasv** [eslint/eslint#2215]
Hello,

I've just setup:
```
    "space-in-brackets": [2, "always", {
      "singleValue": false,
      "objectsInArrays": false,
      "arraysInArrays": false,
      "arraysInObjects": false,
      "objectsInObjects": false,
      "propertyName": false
    }],
```

No I'm getting error when having following code:
```
var test = {
  prop: [someVariable, function () {
    // code...
  }]
}
```
It says `A space is required after '['  space-in-brackets`

Which forces me to write code like this:

```
var test = {
  prop: [ someVariable, function () {
    // code...
  }]
}
```

Now this looks inconsistent and pretty ugly. Is this a bug or some misconfiguration that I'm having?

**Comment by nzakas** [comment]
This is behaving as expected. You have specified that you always want spaces in brackets and then opted out of a few circumstances, none of which cover your example. Do you really want spaces everywhere? With so many exclusions, are you certain you don't want "never" as your setting?

**Comment by bytasv** [comment]
I actually don't care much about Arrays... What I really want is that linter would throw error when objects don't have spacing, i.e.:
```javascript
var a = {val: 123}
```

But I don't see how I could config to that :(

**Comment by nzakas** [comment]
Have you tried setting to "never" and setting all the exceptions to true?

FWIW, #1576 will probably be the long-term answer to this problem.

Also, this rule does not deal with block statement braces, that's why you don't see an error for function blocks.

**Comment by bytasv** [comment]
@nzakas "never" and all true does not work, but it seems that the PR that is pending would solve my issue

**Comment by gyandeeps** [comment]
Going to close this discussion as the problem will be solved by issue: https://github.com/eslint/eslint/issues/2225
